hazelcast/src/main/java/com/hazelcast/query/impl/predicates/AbstractPredicate.java

```diff
@@ -134,11 +138,26 @@ private Comparable convert(AttributeType entryAttributeType, Comparable entryAtt
             } else if (entryAttributeType != null) {
                 return entryAttributeType.getConverter().convert(givenAttributeValue);
             } else {
+                TypeConverter tc = inferTypeConverter(entryAttributeClass);
```

Comment:
To my understanding, the new date types should be registered in `ReflectionHelper.getAttributeType`, that way this custom type inference might be avoided and indexes should pick up the proper type as well.

Reply:
I wasn't aware of this. I try to change the code and update the PR. Thanks